### PR TITLE
react-graph: graph contents are rendered in React

### PR DESCRIPTION
**Pull request by ryanfan**
Will work on adding the logic structure of the graph

### CODE REVIEW

**Comment by Ian-Stewart-Binks** [comment]
@ryanfan Nice. Thanks!

File: public/js/graph/react_node.js.jsx [hunk 1]
```diff
@@ -50,26 +50,61 @@ function getNodes(mode){
...
 }
 
+//Quick fix to initiallyTakeable case-sensitivity
+//Will later remove initiallyTakeable, by making a check if prerequisite list is empty or not
+var initiallyTakeableUpper = $.map(initiallyTakeable, function(item, index) {
+    return item.toUpperCase();
+});
+
 var ReactSVG = React.createClass({
+    componentDidMount: function(){
```
Comment (Ian-Stewart-Binks) [inline public/js/graph/react_node.js.jsx]: `(){` -> `() {`

File: public/js/graph/react_node.js.jsx [hunk 2]
```diff
@@ -50,26 +50,61 @@ function getNodes(mode){
...
+            entry['children'].push(childEntry);
         });
-        dict_list.push(entry);
+        dictList.push(entry);
     });
-    return dict_list;
+    return dictList;
 }
 
+//Quick fix to initiallyTakeable case-sensitivity
```
Comment (david-yz-liu) [inline public/js/graph/react_node.js.jsx]: @ryanfan Just remove the "initially takeable" code for now. I don't care if nothing is initially takeable.

File: public/js/graph/react_node.js.jsx [hunk 3]
```diff
@@ -50,26 +50,61 @@ function getNodes(mode){
...
+        var markerNode = React.findDOMNode(this.refs.marker);
+        
+        svgNode.setAttribute('xmlns','http://www.w3.org/2000/svg');
+        svgNode.setAttribute('xmlns:xlink','http://www.w3.org/1999/xlink');
+        svgNode.setAttribute('xmlns:svg','http://www.w3.org/2000/svg');
+        svgNode.setAttribute('xmlns:dc','http://purl.org/dc/elements/1.1/');
+        svgNode.setAttribute('xmlns:cc','http://creativecommons.org/ns#');
+        svgNode.setAttribute('xmlns:rdf','http://www.w3.org/1999/02/22-rdf-syntax-ns#');
+        
+        markerNode.setAttribute('viewBox', '0 0 10 10');
```
Comment (david-yz-liu) [inline public/js/graph/react_node.js.jsx]: I see a couple of these attributes [here](https://facebook.github.io/react/docs/tags-and-attributes.html). Is this a version issue?

File: public/js/graph/react_node.js.jsx [hunk 4]
```diff
@@ -50,26 +50,61 @@ function getNodes(mode){
...
+        markerNode.setAttribute('refX', 4);
+        markerNode.setAttribute('refY', 5);
+        markerNode.setAttribute('markerUnits', 'strokeWidth');
+        markerNode.setAttribute('orient', 'auto');
+        markerNode.setAttribute('markerWidth', 7);
+        markerNode.setAttribute('markerHeight', 7);
+    },
     render: function() {
+        //not all of these properties are supported in React
+        var svgAttrs = {'width':this.props.width, 'height':this.props.height, 'version':'1.1'};
```
Comment (david-yz-liu) [inline public/js/graph/react_node.js.jsx]: Put spaces after `:`

**Pull request merged**